plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-logo.tsx

```diff
@@ -351,10 +361,17 @@ const LogoEdit = ( {
 		<>
 			<MediaUploadCheck>
 				<MediaUpload
-					onSelect={ onInitialSelectLogo }
+					onSelect={ handleMediaUploadSelect }
 					allowedTypes={ ALLOWED_MEDIA_TYPES }
 					render={ ( { open }: { open: () => void } ) =>
-						cloneElement( logoImg, { onClick: open } )
+						cloneElement( logoImg, {
+							onClick() {
```

Comment:
I know that's subjective, but what do you feel about using an arrow function here? I feel it's more clear (not a strong opinion, though):

```suggestion
							onClick: () => {
```

Reply:
Well l always favor normal function in almost any case except when I need to get access to the `this` instance of an outer context (and it's only for simplicity since I can still get access to the contextual `this` when using normal functions through the `bind` utility)

My reason why I don't like arrow functions is because they are treated as anonymous functions which is very bad for debugging. 
<img width="306" alt="image" src="https://github.com/user-attachments/assets/ab2d1715-431a-4f9b-a15e-866ee141662b">
The same thing happens in the profiler 
<img width="276" alt="image" src="https://github.com/user-attachments/assets/80be28dc-6c87-4a5e-992d-b68f36efbb78">
So how do you know what is the function that you are inspecting or debugging?
This makes bug fixing or performance analysis a real hell just because we are obsessed with the arrow functions ;)!

So due to the way react functions behaves (we don't have to care about the `this` anymore) so arrow functions are not required at all.

Reply:
Makes sense, let's not change it, then.